docs/getting-started/quickstart.md

```diff
@@ -93,157 +117,89 @@ crate-type = ["cdylib"]
+Use the `substreams protogen` command to generate the Rust code to communicate with the protobuf.
 
-## Quickstart examples overview
+```bash
+substreams protogen substreams-ethereum-tutorial.yaml
+```
 
-The relationships for the flow of data are defined in the Substreams manifest. Further information is available in the documentation for [defining complex data strategies](../reference-and-specs/manifests.md) through manifest files.
+### 4. Create Rust library and module handlers
 
-### Ethereum example
+Your Substreams module must contain a Rust library that houses the "module handlers." These handlers are responsible for handling blockchain data injected into the module at runtime. The example module handler extracts the version from the <i>Block</i> object in the `block.ver` property, logs it to the terminal, and assigns it to the `version` field of the [`BasicExampleProtoData`](https://github.com/streamingfast/substreams-ethereum-tutorial/blob/main/proto/basicexample.proto) protobuf.
 
-Clone or download the Ethereum example codebase to get started. The example is located in the official [Substreams Ethereum Tutorial GitHub repository](https://github.com/streamingfast/substreams-ethereum-tutorial).
+To include this example module handler in your module, copy it into a new Rust source code file. Use the filename [`lib.rs`](https://github.com/streamingfast/substreams-ethereum-tutorial/blob/main/src/lib.rs) for this file. Make sure to save the Rust library source code file in the `src` directory, located in the root directory of your Substreams module.
+
+{% code overflow="wrap" %}
 
-Take a moment to explore the codebase. Take note of the chain name used through the different files including the [manifest](https://github.com/streamingfast/substreams-ethereum-tutorial/blob/main/substreams-ethereum-tutorial.yaml), and the [TOML build configuration file](https://github.com/streamingfast/substreams-ethereum-tutorial/blob/main/Cargo.toml).&#x20;
+```rust
+mod pb;
 
-Also take note of the module handler, defined in [`lib.rs`](https://github.com/streamingfast/substreams-ethereum-tutorial/blob/main/src/lib.rs), and the custom protobuf definition in the proto directory named [`basicexample.proto`](https://github.com/streamingfast/substreams-ethereum-tutorial/blob/main/proto/basicexample.proto).
+use pb::basicexample;
 
-{% hint style="info" %}
-**Note**: The module handler in the [`lib.rs`](https://github.com/streamingfast/substreams-ethereum-tutorial/blob/main/src/lib.rs) file for the Ethereum example has code specific to the blockchain. The basic block structure for Ethereum blocks can be viewed in the code excerpt.
-{% endhint %}
+use substreams::{log};
+use substreams_ethereum::{pb as ethpb};
 
-{% code title="src/lib.rs" overflow="wrap" %}
-```rust
 #[substreams::handlers::map]
 fn map_basic_eth(block: ethpb::eth::v2::Block) -> Result<basicexample::BasicExampleProtoData, substreams::errors::Error> {
     log::info!("block.ver: {:#?}", block.ver);
```

Comment:
Let's ensure also that only one log line is printed, the printed values should the block's hash/number

Reply:
Updated in tutorial repo and in the code example in this Quickstart.